service/src/main/java/uk/gov/hmcts/reform/fpl/service/casesubmission/CaseSubmissionTemplateDataGenerationService.java

```diff
@@ -0,0 +1,741 @@
+            .collect(toList());
+    }
+
+    private List<DocmosisApplicant> buildDocmosisApplicants(final List<Element<Applicant>> applicants,
+                                                            final Solicitor solicitor) {
+        return applicants.stream()
+            .map(Element::getValue)
+            .filter(Objects::nonNull)
+            .map(Applicant::getParty)
+            .filter(Objects::nonNull)
+            .map(applicant -> buildApplicant(applicant, solicitor))
+            .collect(toList());
+    }
+
+    private List<DocmosisChild> buildDocmosisChildren(final List<Element<Child>> children) {
+        return children.stream()
+            .map(Element::getValue)
+            .filter(Objects::nonNull)
+            .map(Child::getParty)
+            .filter(Objects::nonNull)
+            .map(this::buildChild)
+            .collect(toList());
+    }
+
+    private List<DocmosisOtherParty> buildDocmosisOthers(final List<Element<Other>> other) {
+        return other.stream()
+            .map(Element::getValue)
+            .filter(Objects::nonNull)
+            .map(this::buildOtherParty)
+            .collect(toList());
+    }
+
+    private List<DocmosisProceeding> buildDocmosisProceedings(final List<Element<Proceeding>> proceedings) {
+        return proceedings.stream()
+            .map(Element::getValue)
+            .filter(Objects::nonNull)
+            .map(this::buildProceeding)
+            .collect(toList());
+    }
+
+    private DocmosisProceeding buildProceeding(final Proceeding proceeding) {
+        return DocmosisProceeding.builder()
+            .onGoingProceeding(getValidAnswerOrDefaultValue(proceeding.getOnGoingProceeding()))
+            .proceedingStatus(getDefaultIfNullOrEmpty(proceeding.getProceedingStatus()))
+            .caseNumber(getDefaultIfNullOrEmpty(proceeding.getCaseNumber()))
+            .started(getDefaultIfNullOrEmpty(proceeding.getStarted()))
+            .ended(getDefaultIfNullOrEmpty(proceeding.getEnded()))
+            .ordersMade(getDefaultIfNullOrEmpty(proceeding.getOrdersMade()))
+            .judge(getDefaultIfNullOrEmpty(proceeding.getJudge()))
+            .children(getDefaultIfNullOrEmpty(proceeding.getChildren()))
+            .guardian(getDefaultIfNullOrEmpty(proceeding.getGuardian()))
+            .sameGuardianDetails(
+                concatenateKeyAndValue(
+                    proceeding.getSameGuardianNeeded(),
+                    proceeding.getSameGuardianDetails()))
+            .build();
+    }
+
+    private DocmosisOtherParty buildOtherParty(final Other other) {
+        final boolean isConfidential = equalsIgnoreCase(other.getDetailsHidden(), YES.getValue());
+        return DocmosisOtherParty.builder()
+            .name(other.getName())
+            .gender(formatGenderDisplay(other.getGender(), other.getGenderIdentification()))
+            .dateOfBirth(other.getDOB() != null ? formatLocalDateToString(parse(other.getDOB()), DATE) : DEFAULT_STRING)
```

Comment:
you have a `formatDateDisplay` method that does this

Reply:
yes, went with this as `other.getDOB` returns string